### ISSUE TITLE
`vim.fs.normalize` swallows whitespace escape characters

### ISSUE THREAD

**Issue by jamestrew** [neovim/neovim#27958]
### Problem

whitespace characters like `\n` and `\t` are being half swallowed by `vim.fs.normalize()`

### Steps to reproduce

`:so %` 
```lua
print(vim.fs.normalize([[/tmp/hello\tworld]])) --> /tmp/hello/tworld
```

### Expected behavior

`vim.fs.normalize([[/tmp/hello\tworld]])` returns the original path `/tmp/hello\tworld` with no modifications

This is the behavior of https://docs.python.org/3/library/os.path.html#os.path.normpath

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2593+ga6b6d036b

### Vim (not Nvim) behaves the same?

no, vim.fs

### Operating system/version

Linux archlinux 6.7.5-arch1-1

### Terminal name/version

kitty 0.31.0 

### $TERM environment variable

tmux-256color

### Installation

build from repo

**Comment by zeertzjq** [comment]
Python 3 behaves like this:
```
>>> import os
>>> print(os.path.normpath(r'/tmp/hello\tworld'))
/tmp/hello\tworld
```

**Comment by zeertzjq** [comment]
> `vim.fs.normalize([[/tmp/hello\tworld]])` returns the original path `/tmp/hello\tworld` with no modifications

Why? Why should be backslash be treated differently if it's followed by `t`?

**Comment by zeertzjq** [comment]
And if you really have a `\t` character in the path then it works as expected:
```vim
:lua print(vim.fs.normalize('/tmp/hello\tworld'))
```
```
/tmp/hello^Iworld
```

**Comment by jamestrew** [comment]
If I'm getting filepath from say an external process (`fd`) or even `vim.api.nvim_buf_get_name`, I need to manually do some string manipulation before using `vim.fs.normalize`?

```lua
local bufname = vim.api.nvim_buf_get_name(4) --> /tmp/hello\tworld
print(vim.fs.normalize(bufname)) --> /tmp/testing/hello/tworld.lua
```

**Comment by zeertzjq** [comment]
No, because `vim.fs.normalize()`'s behavior is correct.

**Comment by zeertzjq** [comment]
If `nvim_buf_get_name` returns `\n` or `\t` then they are literal `\` and `n` or `\` and `t` characters, not whitespace.

**Comment by jamestrew** [comment]
Yeah you're right sorry. Just a bad expectations on my parts.

**Comment by jamestrew** [comment]
This kept bugging me... I definitely confused the issue originally and I apologize for wasting time.
But let me try again in a form of a question first and I can reopen a new issue if it's warranted.

So `vim.fs.normalize` currently mangles valid unix paths with literal backslashes. This makes `vim.fs.normalize` undesirable to use as a catch all for "normalizing" paths. Is this the intended design of `vim.fs.normalize`?

**Comment by zeertzjq** [comment]
@gpanders

**Comment by justinmk** [comment]
> So `vim.fs.normalize` currently mangles valid unix paths with literal backslashes. This makes `vim.fs.normalize` undesirable to use as a catch all for "normalizing" paths. Is this the intended design of `vim.fs.normalize`?

Not intentional. normalize() should be reliable.

**Comment by jamestrew** [comment]
opened a new issue to avoid confusion with this original erroneous issue.
